src/integrationtests/java/com/aws/greengrass/integrationtests/SecretManagerServiceIntegTest.java

```diff
@@ -0,0 +1,196 @@
+        String secretArn = "arn:aws:secretsmanager:us-east-1:999936977227:secret:randomSecret-74lYJh";
+        String secretName = "randomSecret";
+        lenient().doReturn(software.amazon.awssdk.services.secretsmanager.model.GetSecretValueResponse.builder()
+                .name(secretName).arn(secretArn).secretString("secretValue").versionId(VERSION_ID)
+                .versionStages(CURRENT_LABEL)
+                .createdDate(Instant.now().minusSeconds(1000000)).build())
+                .when(secretClient).getSecret(GetSecretValueRequest.builder().secretId(secretArn).versionStage(CURRENT_LABEL).build());
+
+        lenient().doReturn(software.amazon.awssdk.services.secretsmanager.model.GetSecretValueResponse.builder()
+                .name(secretName).arn(secretArn).secretString("secretValue2").versionId("id2")
+                .versionStages("new").createdDate(Instant.now().minusSeconds(1000000)).build())
+                .when(secretClient).getSecret(GetSecretValueRequest.builder().secretId(secretArn).versionStage("new").build());
+    }
+
+    @AfterEach
+    void cleanup() {
+        kernel.shutdown();
+    }
+
+    @BeforeEach
+    void setup(ExtensionContext context) {
+        // Set this property for kernel to scan its own classpath to find plugins
+        System.setProperty("aws.greengrass.scanSelfClasspath", "true");
+        ignoreErrors(context);
+    }
+
+    @Test
+    void GIVEN_secret_service_WHEN_started_with_bad_parameter_config_THEN_starts_successfully(ExtensionContext context)
+            throws Exception {
+        ignoreExceptionOfType(context, java.lang.IllegalArgumentException.class);
+        startKernelWithConfig("badConfig.yaml", State.RUNNING);
+    }
+
+    @Test
+    void GIVEN_secret_service_WHEN_started_without_secrets_THEN_starts_successfully(ExtensionContext context)
+            throws Exception {
+        startKernelWithConfig("emptyParameterConfig.yaml", State.RUNNING);
+    }
+
+    @Test
+    void GIVEN_secret_service_WHEN_started_without_secret_entry_THEN_starts_successfully(ExtensionContext context)
+            throws Exception {
+        startKernelWithConfig("emptySecretConfig.yaml", State.RUNNING);
+    }
+
+    @Test
+    void GIVEN_secret_service_WHEN_request_invalid_THEN_correct_response_returned(ExtensionContext context)
+            throws Exception {
+        ignoreExceptionOfType(context, GetSecretException.class);
+        startKernelWithConfig("config.yaml", State.RUNNING);
+        final String serviceName = "ComponentRequestingSecrets";
+
+        software.amazon.awssdk.aws.greengrass.model.GetSecretValueRequest req =
+                new software.amazon.awssdk.aws.greengrass.model.GetSecretValueRequest();
+        req.setSecretId("");
+        req.setVersionId(VERSION_ID);
+        GreengrassCoreIPCClientV2 clientV2 = IPCTestUtils.connectV2Client(kernel, serviceName);
+        assertThrows(ServiceError.class, () -> clientV2.getSecretValue(req));
```

Comment:
assert something about the message being reasonable.

Reply:
This one will be updated later as it has to raise something like an `InvalidRequestException` instead of a service error.

Reply:
You cannot add new error types to an existing API FYI. It is not backward compatible. If that exception type is already declared for this API then that's OK.

Reply:
Updated with the message. 

I can still include `InvalidArgumentsError` here when we update that, right? https://github.com/awslabs/smithy-iot-device-sdk-greengrass-ipc/blob/main/greengrass-ipc-model/main.smithy#L208

Reply:
No you cannot, that's my point. Adding a new error type is not backward compatible.

Reply:
Yes,  I was confused, it is safe to add an exception type that already exists in the smithy model to an existing API. Adding a new error type to an existing API is not safe.